src/internal/resources/manager.cpp

```diff
@@ -134,6 +148,18 @@ Manager::Manager(std::unique_ptr<system::Resources> resources) :
         m_partitions.emplace_back(
             m_runnable.at(host_partition_id), i, m_host.at(host_partition_id), m_device.at(i), m_network.at(i));
     }
+
+    // set thread local access to resources on all fiber task queues and any future thread created by the runtime
+    for (auto& partition : m_partitions)
+    {
+        m_system->register_thread_local_initializer(partition.partition().host().cpu_set(), [this, &partition] {
+            m_thread_resources = this;
+            if (system().partitions().device_to_host_strategy() == PlacementResources::Dedicated)
+            {
+                m_thread_partition = &partition;
```

Comment:
What is the significance of this; what would we observe if PlacementResources is something other than dedicated

Reply:
This is the primary concern of #114. 

My vote to to remove shared and dedicated and keep only dedicated functionality. 

The edge cases where you have more GPUs than cores we could either allow oversubscription or limit the number of devices include in the runtime resources.